src/docs/development/ui/assets-and-images.md

```diff
@@ -233,6 +233,10 @@ awareness when loading images. (If you work with some of the lower
 level classes, like [`ImageStream`][] or [`ImageCache`][],
 you'll also notice parameters related to scale.)
 
+{{site.alert.note}}
+  [device pixel ratio][] depends on [MediaQueryData.size][] which requires to have either [MaterialApp][] or [CupertinoApp][] as ancestor of your [`AssetImage`][]
```

Comment:
```suggestion
  [device pixel ratio][] depends on [MediaQueryData.size][] which requires to have either a [MaterialApp][] or [CupertinoApp][] as an ancestor of your [`AssetImage`][]
```

Reply:
Maybe this need a period at the end of the sentence.

Reply:
add both changes